### PR TITLE
compiler: prevent multisubdimension expressions duplicates

### PR DESCRIPTION
**Pull request by mloubout**
Can't easily make a test for it here. Breaks `abox` with extra dims

### CODE REVIEW

**Review by FabioLuporini — APPROVED** [review]
So a test is being added to PRO right?

Approved!

File: devito/passes/clusters/implicit.py [hunk 1]
```diff
@@ -106,6 +107,13 @@ def callback(self, clusters, prefix):
             # The "implicit expressions" created for the MultiSubDomain
             exprs, dims, sub_iterators = make_implicit_exprs(d.msd, c)
 
+            # Make sure the "implicit expressions" aren't scheduled in
+            # an inner loop. E.g schedule both for `t, xi, yi` and `t, d, xi, yi`
```
Comment (FabioLuporini) [inline devito/passes/clusters/implicit.py:111]: what do you mean by "both" here? 
I think the example could be a bit more clear here
OK in another PR

File: devito/passes/clusters/implicit.py [hunk 2]
```diff
@@ -106,6 +107,13 @@ def callback(self, clusters, prefix):
...
+            edims = set(retrieve_dimensions(exprs, deep=True))
+            if dim not in edims and any(d.dim in edims for d in prefix):
```
Comment (FabioLuporini) [inline devito/passes/clusters/implicit.py:113]: Actually, `dim` is in prefix, so the following should suffice:

```
if any(d.dim in edims for d in prefix):
```

In fact, the following might work as well:

```
if prefix.itdims & edims:
```

File: devito/passes/clusters/implicit.py (hunk 1 above)
Comment (mloubout) [reply devito/passes/clusters/implicit.py:111]: For clusters with itspace `t, xi, yi` and `t, d, xi, yi` then this statement will be reached twice during the `self.process` once for `t` as a prefix and once for `t, d`

**Pull request merged**